### ISSUE TITLE
Mailtrain v2 crashes with latest git commit

### ISSUE THREAD

**Issue by trantienthanh** [Mailtrain-org/mailtrain#665]
git commit bb237b3 Fix - URL bases replacement didn't work for HBS tag language.

ERR!    'Error: The migration directory is corrupt, the following files are missing: 20190726150000_shorten_field_column_names.js\n    at validateMigrationList (/opt/mailtrain/server/node_modules/knex/lib/migrate/Migrator.js:341:11)\n    at migrationListResolver.listAllAndCompleted.tap.value (/opt/mailtrain/server/node_modules/knex/lib/migrate/Migrator.js:77:91)\n    at PassThroughHandlerContext.finallyHandler (/opt/mailtrain/server/node_modules/bluebird/js/release/finally.js:57:23)\n    at PassThroughHandlerContext.tryCatcher (/opt/mailtrain/server/node_modules/bluebird/js/release/util.js:16:23)\n    at Promise._settlePromiseFromHandler (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:517:31)\n    at Promise._settlePromise (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:574:18)\n    at Promise._settlePromise0 (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:619:10)\n    at Promise._settlePromises (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:699:18)\n    at Promise._fulfill (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:643:18)\n    at PromiseArray._resolve (/opt/mailtrain/server/node_modules/bluebird/js/release/promise_array.js:126:19)\n    at PromiseArray._promiseFulfilled (/opt/mailtrain/server/node_modules/bluebird/js/release/promise_array.js:144:14)\n    at Promise._settlePromise (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:579:26)\n    at Promise._settlePromise0 (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:619:10)\n    at Promise._settlePromises (/opt/mailtrain/server/node_modules/bluebird/js/release/promise.js:699:18)\n    at _drainQueueStep (/opt/mailtrain/server/node_modules/bluebird/js/release/async.js:138:12)\n    at _drainQueue (/opt/mailtrain/server/node_modules/bluebird/js/release/async.js:131:9)\n    at Async._drainQueues (/opt/mailtrain/server/node_modules/bluebird/js/release/async.js:147:5)\n    at Immediate.Async.drainQueues [as _onImmediate] (/opt/mailtrain/server/node_modules/bluebird/js/release/async.js:17:14)\n    at runCallback (timers.js:705:18)\n    at tryOnImmediate (timers.js:676:5)\n    at processImmediate (timers.js:658:5)\n    at process.topLevelDomainCallback (domain.js:126:23)' }

**Comment by bures** [comment]
Yep. That the unfortunate 20190726150000_shorten_field_column_names.js migration. It was just flawed and not really needed. And in top of it, it likely messed up up you segments. I'll issue a script which will fix the DB for those who already applied this migration. Sorry for the trouble guys.

**Comment by bures** [comment]
Try the latest commit. You will have to run:
```NODE_ENV=production node fix-20190726150000_shorten_field_column_names.js```
in `server/setup/knex/fixes`.

**Comment by gkisdaroczi** [comment]
Hello Tomas,

I tried to run the fix, but there is something.

> Try the latest commit. You will have to run:
> `NODE_ENV=production node fix-20190726150000_shorten_field_column_names.js`
> in `server/setup/knex/fixes`.

I did it on development-tb branch and as you said I went into the directory and run the command.
Am I miss something? :)

There is the config.js. And knexfile.js in /server/setup/knex.

Thank you for your response! :)

Best regards, Gabor

bash-4.4# cd server/setup/knex/fixes
bash-4.4# NODE_ENV=production node fix-20190726150000_shorten_field_column_names.js
internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module 'server/setup/knex/config'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/app/server/setup/knex/config.js:7:16)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)

**Comment by bures** [comment]
You have too do it in development branch. There is no fixes directory in development-tb as far as I can tell.

**Comment by gkisdaroczi** [comment]
I know that. :) I have just copied that ;P

**Comment by gkisdaroczi** [comment]
Ok. It's almost done. :)
I just changed branch to development add missing migration js to server/setup/knex/fixes from development-tb branch and now just started the mailtrain without errors. Hooray! :)

But now I don't know what I have to do.
I started a new hungarian-translation branch from development-tb only on local.
But because that is not up-to-date, which branch I have to use to continue the development? :/

Thanks,
